### PR TITLE
Feature/ops cg cpp

### PR DESCRIPTION
**Pull request by Ashutosh-Londhe**

### CODE REVIEW

File: apps/fortran/multiDim/multidim.F90 [hunk 1]
```diff
@@ -37,50 +37,43 @@
 program MULTIDIM
   use OPS_Fortran_Reference
   use OPS_CONSTANTS
-  !use OPS_Fortran_Declarations
```
Comment (reguly) [inline apps/fortran/multiDim/multidim.F90:40]: Remove commented lines

File: apps/fortran/poisson/poisson.F90 [hunk 2]
```diff
@@ -49,43 +49,43 @@ program POISSON
 
   implicit none
 
-  !integer logical_size_x
```
Comment (reguly) [inline apps/fortran/poisson/poisson.F90:52]: no need for these lines

File: makefiles/Makefile.common [hunk 3]
```diff
@@ -33,13 +33,55 @@ CUDART=-lcudart
 CURAND=-lcurand
 HIPRAND=-lhiprand
 
-COMPILER_AVAILABLE := clang gnu intel icx intel_win mic pgi cray xl hip xlp8 aocc
-
-ifeq ($(wildcard $(OPS_INSTALL_PATH)/../makefiles/Makefile.$(OPS_COMPILER)),)
-print:
-	@echo "unrecognised value for OPS_COMPILER: "$(OPS_COMPILER)
+ifeq ($(OPS_COMPILER),clang)
```
Comment (reguly) [inline makefiles/Makefile.common:36]: Let's not bring this back. Revert to the old one, unless there is a good reason

**Review by reguly — CHANGES_REQUESTED** [review]

File: apps/c/adi/Makefile.old
[the remark this answers is not in the document]
Comment (Ashutosh-Londhe) [reply apps/c/adi/Makefile.old:1]: not required. removed now.

File: apps/c/adi_burger_3D/Makefile.old
[the remark this answers is not in the document]
Comment (Ashutosh-Londhe) [reply apps/c/adi_burger_3D/Makefile.old:1]: not required. removed now.

File: makefiles/Makefile.common (hunk 3 above)
Comment (Ashutosh-Londhe) [reply makefiles/Makefile.common:36]: the latest makefile doesnt have this. maybe ur looking it before rebase??

File: apps/c/mgrid/mgrid.cpp [hunk 4]
```diff
@@ -200,12 +200,14 @@ int main(int argc, const char **argv)
   ops_halo_transfer(halos[0]);
   ops_halo_transfer(halos[1]);
 
+  int size4_0 = size4[0];
+  int size4_1 = size4[1];
   ops_par_loop(prolong_check, "prolong_check", grid0, 2, iter_range_large,
                ops_arg_dat(data5, 1, S2D_5pt, "double", OPS_READ),
                ops_arg_idx(),
                ops_arg_reduce(reduct_err, 1, "int", OPS_MAX),
-               ops_arg_gbl(&size4[0], 1, "int", OPS_READ),
```
[the remark this answers is not in the document]
Comment (Ashutosh-Londhe) [reply apps/c/mgrid/mgrid.cpp:207]: was earlier when this was added. now not required. reverting to old code.

File: makefiles/Makefile.common (hunk 3 above)
Comment (reguly) [reply makefiles/Makefile.common:36]: https://github.com/OP-DSL/OPS/blob/703b27a3330cd4c79e78c31434f5f641dfb47218/makefiles/Makefile.common#L36
I am saying we should not have the ugly ifeq structure, instead the wildcard stuff

Comment (Ashutosh-Londhe) [reply makefiles/Makefile.common:36]: ok corrected it

**Review by reguly — APPROVED** [review]

**Pull request merged**